File: python/1014. Best Sightseeing Pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        runningMax = values[-1] - (len(values)-1)
        answer = float('-inf')
        for i in range(len(values)-2, -1, -1):
            answer = max(answer, runningMax + values[i] + i)
            runningMax = max(runningMax, values[i] - i)
        return answer

File: java/1014. Best Sightseeing Pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int answer = Integer.MIN_VALUE, runningMax = values[values.length-1] - (values.length-1);
        for (int i = values.length-2; i >= 0; i--) {
            answer = Math.max(answer, values[i] + i + runningMax);
            runningMax = Math.max(values[i] - i, runningMax);
        }
        return answer;
    }
}

